x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/TextSimilarityProcessor.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.inference.nlp;
+
+import org.elasticsearch.ElasticsearchStatusException;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.xpack.core.ml.inference.results.InferenceResults;
+import org.elasticsearch.xpack.core.ml.inference.results.TextSimilarityInferenceResults;
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.NlpConfig;
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.TextSimilarityConfig;
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.Tokenization;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+import org.elasticsearch.xpack.ml.inference.nlp.tokenizers.NlpTokenizer;
+import org.elasticsearch.xpack.ml.inference.nlp.tokenizers.TokenizationResult;
+import org.elasticsearch.xpack.ml.inference.pytorch.results.PyTorchInferenceResult;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+import static org.elasticsearch.xpack.core.ml.inference.trainedmodel.InferenceConfig.DEFAULT_RESULTS_FIELD;
+
+public class TextSimilarityProcessor extends NlpTask.Processor {
+
+    TextSimilarityProcessor(NlpTokenizer tokenizer, TextSimilarityConfig config) {
+        super(tokenizer);
+    }
+
+    @Override
+    public void validateInputs(List<String> inputs) {
+        // nothing to validate
+    }
+
+    @Override
+    public NlpTask.RequestBuilder getRequestBuilder(NlpConfig nlpConfig) {
+        if (nlpConfig instanceof TextSimilarityConfig textSimilarityConfig) {
+            return new RequestBuilder(tokenizer, textSimilarityConfig.getText());
+        }
+        throw ExceptionsHelper.badRequestException(
+            "please provide configuration update for text_similarity task including the desired [text]"
+        );
+    }
+
+    @Override
+    public NlpTask.ResultProcessor getResultProcessor(NlpConfig nlpConfig) {
+        if (nlpConfig instanceof TextSimilarityConfig textSimilarityConfig) {
+            String resultsFieldValue = textSimilarityConfig.getResultsField();
```

Comment:
Is there a reason this was assigned in a local variable instead of passed straight into the `ResultProcessor` ctor?